### PR TITLE
Clean code for several recent modifications

### PR DESCRIPTION
**Pull request by stweil**

### CODE REVIEW

File: src/ccmain/thresholder.cpp [hunk 1]
```diff
@@ -205,22 +204,19 @@ std::tuple<bool, Image, Image, Image> ImageThresholder::Threshold(
...
+  if (method == ThresholdMethod::TiledSauvola) {
     r = pixSauvolaBinarizeTiled(pix_grey, 25, 0.40, 300, 300, pix_thresholds,
                                 pix_binary);
+  } else {
+    // AdaptiveOtsu.
+    r = pixOtsuAdaptiveThreshold(pix_grey, 300, 300, 0, 0, 0.1,
+                                 pix_thresholds, pix_binary);
   }
 
-  bool ok = r == 0 ? true : false;
```
Comment (amitdo) [inline src/ccmain/thresholder.cpp:223]: I intend to add more options from Leptonica, but I guess I can keep this the last option.

Comment (stweil) [reply src/ccmain/thresholder.cpp:223]: That should not be a problem. You can also choose any order of the different methods. Just make sure that the last block is executed unconditionally. Then the compilers are happy.

File: src/api/baseapi.cpp [hunk 2]
```diff
@@ -2107,15 +2107,12 @@ bool TessBaseAPI::Threshold(Pix **pix) {
...
-  auto pageseg_mode = static_cast<PageSegMode>(static_cast<int>(tesseract_->tessedit_pageseg_mode));
-
-  Image pix_binary(*pix);
-  Image pix_grey;
-  Image pix_thresholds;
 
   auto thresholding_method = static_cast<ThresholdMethod>(static_cast<int>(tesseract_->thresholding_method));
 
   if (thresholding_method == ThresholdMethod::Otsu) {
+    auto pageseg_mode = static_cast<PageSegMode>(static_cast<int>(tesseract_->tessedit_pageseg_mode));
```
Comment (amitdo) [inline src/api/baseapi.cpp:2114]: You can drop it, it is not used in  ThresholdToPix().

File: src/ccmain/thresholder.cpp [hunk 3]
```diff
@@ -190,11 +190,6 @@ std::tuple<bool, Image, Image, Image> ImageThresholder::Threshold(
   Image pix_binary = nullptr;
   Image pix_thresholds = nullptr;
 
-  if (image_width_ > INT16_MAX || image_height_ > INT16_MAX) {
```
Comment (amitdo) [inline src/ccmain/thresholder.cpp:194]: 1) As I said elsewhere, this block was copied from another function in the same file. You should delete the other block too.

Comment (amitdo) [reply src/ccmain/thresholder.cpp:194]: There might be security issues in allowing unlimited file size.

You should check if Leptonica has any image size limits.

**Review by egorpugin — APPROVED** [review]

File: src/ccmain/thresholder.cpp [hunk 4]
```diff
@@ -205,22 +204,19 @@ std::tuple<bool, Image, Image, Image> ImageThresholder::Threshold(
...
 
   int r;
-  if (method == ThresholdMethod::AdaptiveOtsu) {
-    r = pixOtsuAdaptiveThreshold(pix_grey, 300, 300, 0, 0, 0.1,
-                                 pix_thresholds, pix_binary);
-  } else if (method == ThresholdMethod::TiledSauvola) {
+  if (method == ThresholdMethod::TiledSauvola) {
     r = pixSauvolaBinarizeTiled(pix_grey, 25, 0.40, 300, 300, pix_thresholds,
                                 pix_binary);
+  } else {
```
Comment (amitdo) [inline src/ccmain/thresholder.cpp:208]: Maybe print a warning message when method >= ThresholdMethod::Max ?

**Pull request merged**